src/librustc/lint/context.rs

```diff
@@ -452,8 +452,7 @@ pub fn raw_struct_lint<'a>(sess: &'a Session,
     }
 
     if let Some(span) = def {
-        let explanation = "lint level defined here";
-        err.span_note(span, &explanation);
+        sess.diag_span_note_once(&mut err, span, "lint level defined here");
```

Comment:
Hmm. So the key is the span+message, but it's possible that the level for many lints is set at one point (e.g., `#[allow(warnings)]`). Do we want to issue one note per kind of lint, or just one per span? I'm actually not sure =)